core/matrix/dense.cpp

```diff
@@ -1158,6 +1159,41 @@ void Dense<ValueType>::row_gather_impl(const Dense<ValueType>* alpha,
 }
 
 
+template <typename ValueType>
+template <typename OutputType, typename IndexType>
+void Dense<ValueType>::row_scatter_impl(const array<IndexType>* row_idxs,
+                                        Dense<OutputType>* target) const
+{
+    auto exec = this->get_executor();
+    dim<2> expected_dim{row_idxs->get_num_elems(), this->get_size()[1]};
+    GKO_ASSERT_EQUAL_DIMENSIONS(expected_dim, this);
+    GKO_ASSERT_EQUAL_COLS(this, target);
```

Comment:
it is also checked by the `EQUAL_DIMENSIONS`

Reply:
the first one checks `expected_dim` and `this`, while the second one checks `this` and `target`. So, I don't think it's covered.

Reply:
you are right.